rubix-core/src/main/java/com/qubole/rubix/core/ReadRequestChainFactory.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) 2019. Qubole Inc
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. See accompanying LICENSE file.
+ */
+package com.qubole.rubix.core;
+
+import com.qubole.rubix.spi.BookKeeperFactory;
+import com.qubole.rubix.spi.thrift.BookKeeperService;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileSystem;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+public final class ReadRequestChainFactory
+{
+  private ReadRequestChainFactory()
+  { }
+
+  public static <T extends ReadRequestChain> T createReadRequestChain(Class<T> readRequestChainType, FSDataInputStream inputStream)
```

Comment:
Rather than taking a Class and then comparing it later, have createRemoteReadRequestChain(), createCachedReadRequestChain(), etc methods here

Reply:
Done